adagrams/game.py

```diff
@@ -1,11 +1,186 @@
+    letter_bank_dict = {}
+    for letter in letter_bank:
+        if letter in letter_bank_dict:
+            letter_bank_dict[letter] += 1
+        else:
+            letter_bank_dict[letter] = 1
+
+    #build hand_dict
+    hand_dict = {}
+    for letter in word.upper():
+        if letter in hand_dict:
+            hand_dict[letter] += 1
+        else:
+            hand_dict[letter] = 1
+    
+    #compare two dictionaries
+    for letter in hand_dict:
+        if letter not in letter_bank_dict:
+            return False
+        elif hand_dict[letter] > letter_bank_dict[letter]:
+            return False
+    return True
+
 
 def score_word(word):
-    pass
+    word_score = 0
+    if len(word) >= 7:
+        word_score += 8
+    for letter in word.upper():
+        letter_score = letter_score_dict[letter]
+        word_score += letter_score
+    return word_score
 
+# def get_highest_word_score(word_list):
+#     word_score_dict = {} 
+#     highest_score_yet = 0
+#     highest_word_yet = ""
+#     highest_len_yet = 0
+#     for word in word_list:
+#         word_score_dict[word] = score_word(word)
+#     for word, score in word_score_dict.items():
+#         if score > highest_score_yet:
+#             highest_score_yet = score
+#             highest_word_yet = word
+#             highest_len_yet = len(word)
+#         if score == highest_score_yet:
+#             if len(word) == 10:
+#                 highest_score_yet = score
+#                 highest_word_yet = word
+#                 highest_len_yet = len(word)
+#                 winning_tuple = (highest_word_yet, highest_score_yet)
+#             elif len(word) < highest_len_yet:
+#                 highest_score_yet = score
+#                 highest_word_yet = word
+#                 highest_len_yet = len(word)
+#     winning_tuple = (highest_word_yet, highest_score_yet)
+#     return winning_tuple
+    
 def get_highest_word_score(word_list):
-    pass
\ No newline at end of file
+    highest_word_score = 0
+    highest_score_words = []
+    #find the highest score
+    for word in word_list:
+        if score_word(word) > highest_word_score:
+            highest_word_score = score_word(word)
+    #add word(s) with the highest score to the highest_word_score_words list
+    for word in word_list:
+        if score_word(word) == highest_word_score:
+            highest_score_words.append(word)
+    # return the word (as a tuple) with the highest score if theres only one
+    if len(highest_score_words) == 1:
+        return tuple([str(highest_score_words[0]), score_word(highest_score_words[0])])
+    # break tie
+    elif len(highest_score_words) > 1:
+        # initialize the shortest length
+        shortest_word_len = 9
+        # find the first 10 or shortest length
+        for word in highest_score_words:            
+            # return the first 10 letter long word
+            if len(word) == 10:
+                return tuple([word, score_word(word)])    
```

Comment:
I might pull the 10-letter check out into its own loop. From a complexity standpoint, it doesn't matter whether we do this check in an existing loop, or in a separate loop. And it might be a little easier to reason about if it were handled on its own rather than forcing the introduction of an `if`/`elif` here.